wwwroot/src/js/wiki.js

```diff
@@ -131,23 +131,69 @@ function buildQuery(input)
     return output.join('&');
 }
 
+function hasChanges(changed, formData) {
+    var Form = document.getElementById('TheInternet');
+    return changed || (new FormData(Form) !== formData);
+}
+
 $(document).ready(function()
 {
     // Populate saved name from local storage
     var name = localStorage.getItem('name');
+    // Check if the title is "Preview:", which means changes are likely pending
+    var title = document.querySelector('div.title');
+    var changed = title && title.textContent.trim().startsWith('Preview:');
+    // Stash the form so we can use it to compare for changes later
+    var Form = document.getElementById('TheInternet');
+    var formData = new FormData(Form);
 
     if(name)
     {
         $('#Name').value(name);
     }
 
+    $('a').on('click', function(e) {
+        if (hasChanges(changed, formData)) {
+            if (!confirm('You have unsaved changes. Are you sure you want to leave?')) {
+                e.preventDefault();
+            } else {
+                // Override function to make sure it doesn't fire anyways
+                $(window).on('beforeunload', function (e) {});
+                changed = false;
+            }
+        }
+    });
+
+    $(window).on('beforeunload', function(e) {
+        if (hasChanges(changed, formData)) {
+            var confirmationMessage = 'You have unsaved changes. Are you sure you want to leave?';
+            (e || window.event).returnValue = confirmationMessage;
+            return confirmationMessage;
+        }
+    });
```

Comment:
This return message logic doesn't seem to be working? The beforeunload event is working, but I'm getting a generic warning message from my browser, not the custom message defined in the code.

Tested in both Firefox:
![image](https://github.com/user-attachments/assets/bea3ec49-0804-4796-9a17-f99a9b182719)

And Chrome:
![image](https://github.com/user-attachments/assets/eb6aed0c-6eda-4911-9762-e5fd95a5ef72)